pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -33,7 +33,9 @@ class TrainingTypePlugin(Plugin, ABC):
     def __init__(self) -> None:
         self._model = None
         self._results = None
-        self.global_rank = 0
+        self.global_rank = None
+        self.local_rank = None
+        self.world_size = None
```

Comment:
```suggestion
        self.global_rank = 0
        self.local_rank = 0
        self.world_size = 1
```
Can we keep it as before and not set None?

Reply:
Hey @awaelchli. I actually think there is a bug behind this default, which could explain memory on gpu 0 when training on [1, 2] for example.
I will revert this back in the PR, but worth to investigate if everything is fine.